xayn-ai/src/coi/system.rs

```diff
@@ -214,6 +223,160 @@ impl CoiSystem {
             })
             .collect()
     }
+
+    /// Selects the most relevant key phrases for the coi.
+    ///
+    /// The most relevant key phrases are selected from the set of key phrases of the coi and the
+    /// candidates. The computed relevances are a relative score from the interval `[0, 1]`.
+    #[allow(dead_code)]
+    fn select_key_phrases<
+        CP: CoiPoint + CoiPointKeyPhrases,
+        F: Fn(&str) -> Result<Embedding, Error>,
+    >(
+        &self,
```

Comment:
Is better if the logic of the function does not depends on `self` but takes every parameter it needs. it will be easier to test since we don't have to construct auxiliary data structure. Then we can use it here by passing the values from the config.

Reply:
currently all functions and tests of this module are set up with self as the receiver, that's why i implemented this method in the same style. we could rearrange that, but we should do it consistently for all functions of this module, and probably in a different pr.

Reply:
Yes can be done a bit everywhere indeed. In general we should go in that direction because it will make the business logic easier to test, in particular in this module we have lots of functions that are depending only on one or two parameters from the configuration so is not much overhead to write them explicitly. We also have more control of which parameter from the configuration the function is using.

Reply:
But ok let's do it in another pr or when we move the code.